### PR TITLE
feat[chart]: Add chaos-charts for CPU-Hog experiment

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <uditgaurav@gmail.com>

**_Add chaos-charts for CPU-Hog experiment_**

This PR contains the chaos-chart for CPU-Hog experiment. This experiment is used to inject a CPU spike chaos on a Kubernetes Node - which will help us to read the scenario where the node consumes the CPU equal to the CPU capacity to that node.

Folder added:
- [X] chaos-charts/charts/generic/CPU-Hog

FIles added:
- [X] cpu-hog.chartserviceversion.yaml
- [X] experiment.yaml

Logs :
[logs.txt](https://github.com/litmuschaos/chaos-charts/files/3823465/logs.txt)

### CODE REVIEW

File: charts/generic/cpu-hog/experiment.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apiVersion: litmuschaos.io/v1alpha1
```
Comment (ksatchit) [inline charts/generic/cpu-hog/experiment.yaml:1]: Lets add the right permissions

Comment (uditgaurav) [reply charts/generic/cpu-hog/experiment.yaml:1]: ok

File: charts/generic/cpu-hog/experiment.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        - "*"
+    image: "litmuschaos/ansible-runner:ci"
+    args:
+    - -c
+    - ansible-playbook ./experiments/generic/cpu_hog/cpu_hog_ansible_logic.yml -i /etc/ansible/hosts -vv; exit 0
+    command:
+    - /bin/bash
+    env:
+    - name: ANSIBLE_STDOUT_CALLBACK
+      value: 'default'
```
Comment (ispeakc0de) [inline charts/generic/cpu-hog/experiment.yaml:40]: can you please provide the space after each env ?

File: charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+apiVersion: litmuchaos.io/v1alpha1
+kind: ChartServiceVersion
+metadata:
+  name: cpu_hog
```
Comment (chandankumar4) [inline charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml]: It should be `cpu-hog`

File: charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  version: 0.1.3
+  annotations:
+    categories: Kubernetes
+    vendor: CNCF
+    createdAt: 2019-08-12T10:28:08Z
+    repository: https://github.com/litmuschaos/chaos-charts
+    support: https://slack.kubernetes.io/
+spec:
+  displayName: cpu_hog
```
Comment (chandankumar4) [inline charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml]: It should be `cpu-hog`

File: charts/generic/cpu-hog/experiment.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      value: '60'
+
+    - name: PLATFORM
+      value: 'ANY'
+
+    - name: LIB
+      value: ''
+      
+    labels:
+      name: cpu_hog
```
Comment (chandankumar4) [inline charts/generic/cpu-hog/experiment.yaml]: It should be `cpu-hog`

File: charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  categoryDescription: |
+    CPU hog contains chaos to disrupt state of kubernetes resources. Experiments can inject a cpu skipe on a node where the application pod is scheduled.
+    - Causes (forced/graceful) cpu hog on a perticular node where the application deployment is available.
+    - After test the recovery should be manual of the application pod and node in case they are not in appropriate state.
+ 
+  keywords:
+    - Kubernetes
+    - CPU
+    - State
```
Comment (chandankumar4) [inline charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml:22]: Can you add the platform also?

File: charts/generic/cpu-hog/experiment.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    permissions:
+      apiGroups:
+        - ""
+        - "extensions"
+        - "apps"
+        - "batch"
+        - "litmuschaos.io"
+      resources:
+        - "daemonsets"
+        - "deployments"
```
Comment (ksatchit) [inline charts/generic/cpu-hog/experiment.yaml:21]: statefulsets

File: charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    CPU hog contains chaos to disrupt state of kubernetes resources. Experiments can inject a cpu skipe on a node where the application pod is scheduled.
+    - Causes (forced/graceful) cpu hog on a perticular node where the application deployment is available.
+    - After test the recovery should be manual of the application pod and node in case they are not in appropriate state.
+ 
+  keywords:
+    - Kubernetes
+    - CPU
+    - State 
+  platforms:
+    - gke
```
Comment (chandankumar4) [inline charts/generic/cpu-hog/cpu-hog.chartserviceversion.yaml]: GKE

File: charts/generic/cpu-hog/experiment.yaml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    env:
+
+    - name: ANSIBLE_STDOUT_CALLBACK
+      value: 'default'
+
+    - name: TOTAL_CHAOS_DURATION
+      value: '60'
+
+    - name: PLATFORM
+      value: 'ANY'
```
Comment (chandankumar4) [inline charts/generic/cpu-hog/experiment.yaml:46]: Does it support all platforms or only GKE?

**Review by chandankumar4 — APPROVED** [review]
LGTM

**Review by ksatchit — APPROVED** [review]

**Pull request merged**